memstatsd/memstatsd.go

```diff
@@ -205,7 +230,16 @@ func joinTags(tags ...map[string]string) string {
 	}
 	var str string
 	for k, v := range tags[0] {
-		str += fmt.Sprintf(",%s=%s", k, v)
+		str += fmt.Sprintf(",%s=%s", strings.ToLower(k), v)
 	}
 	return str
 }
+
+func structToMap(s interface{}) map[string]string {
+	var m map[string]string
+
+	sm, _ := json.Marshal(s)
```

Comment:
While I don't love doing it explicitly ... i think in this case we should avoid json marshal/unmarshal ... would prefer to not have that overhead on every call...even if that pattern exists everywhere else...

Reply:
I got your point. I should be able to accomplish the same with reflection although I think using reflect here is also overkill.